src/Host/Client/Impl/rtvs/R/util.R

```diff
@@ -204,11 +204,17 @@ export_to_pdf <- function(device_id, plot_id, width, height) {
 }
 
 # Helper to publish rmarkdown files remotely
-rmarkdown_publish <- function(blob_id, output_format, encoding) {
+rmarkdown_publish <- function(work_folder, blob_id, output_format, encoding) {
     # Create temp file to store the rmarkdown file
-    rmdpath <- tempfile('rmd_', fileext = '.rmd');
-    on.exit(unlink(rmdpath));
-    writeBin(get_blob(blob_id), rmdpath);
+    if(identical(work_folder, '')) {
+        work_folder <- tempdir();
+    } else {
+        work_folder <- normalizePath(work_folder);
+    }
+
+    input_path <- tempfile('rmd_', tmpdir = work_folder, fileext = '.rmd');
+    writeBin(get_blob(blob_id), input_path);
+    on.exit(file.remove(input_path), add = TRUE);
```

Comment:
This also fixes problem that one of the temp files was not deleted. `on.exit` without `add = TRUE` overwrites previously set actions.